plugins/ingest-script/src/main/java/org/elasticsearch/ingest/script/ScriptProcessor.java

```diff
@@ -0,0 +1,121 @@
+        this.field = field;
+        this.script = script;
+        this.scriptService = scriptService;
+        this.clusterService = clusterService;
+    }
+
+    @Override
+    public void execute(IngestDocument ingestDocument) throws Exception {
+        Map<String, Object> data = new HashMap<>();
+        Map<String, Object> context = new HashMap<>();
+        data.put("ctx", context);
+        context.put("_source", ingestDocument.getSourceAndMetadata());
+        CompiledScript compiledScript = scriptService.compile(script, ScriptContext.Standard.INGEST, emptyMap(), clusterService.state());
+        Object returnValue = scriptService.executable(compiledScript, data).run();
+        ingestDocument.setFieldValue(field, returnValue);
+    }
+
+    @Override
+    public String getType() {
+        return TYPE;
+    }
+
+    public static final class Factory extends AbstractProcessorFactory<ScriptProcessor> {
+
+        private Node node;
+
+        public Factory(Node node) {
+            this.node = node;
+        }
+
+        @Override
+        protected ScriptProcessor doCreate(String processorTag, Map<String, Object> config) throws Exception {
+            String field = readStringProperty(TYPE, processorTag, config, "field");
+            String lang = readOptionalStringProperty(TYPE, processorTag, config, "lang");
+            String id = readOptionalStringProperty(TYPE, processorTag, config, "id");
+            String file = readOptionalStringProperty(TYPE, processorTag, config, "file");
+            String inline = readOptionalStringProperty(TYPE, processorTag, config, "inline");
+
+            boolean containsNoScript = !Strings.hasLength(file) && !Strings.hasLength(id) && !Strings.hasLength(inline);
+            if (containsNoScript) {
+                throw new ElasticsearchException("Need [file], [id] or [inline] parameter to refer to scripts");
+            }
+
+            boolean moreThanOneConfigured = (Strings.hasLength(file) && Strings.hasLength(id)) ||
+                                            (Strings.hasLength(file) && Strings.hasLength(inline)) ||
+                                            (Strings.hasLength(id) && Strings.hasLength(inline));
+            if (moreThanOneConfigured) {
+                throw new ElasticsearchException("Only one of [file], [id] or [inline] may be configured");
+            }
+
+            final Script script;
+            if (Strings.hasLength(file)) {
+                script = new Script(file, FILE, lang, emptyMap());
+            } else if (Strings.hasLength(inline)) {
+                script = new Script(inline, INLINE, lang, emptyMap());
+            } else if (Strings.hasLength(id)) {
+                script = new Script(id, STORED, lang, emptyMap());
+            } else {
+                throw new ElasticsearchException("Could not initialize script");
+            }
+
+            ScriptService scriptService = node.injector().getInstance(ScriptService.class);
```

Comment:
I agree. This dependency should be passed into the `ScriptProcessor.Factory`

I don't think this is better: https://github.com/elastic/elasticsearch/pull/18193/files#diff-509dbced5cd63de8c1bc2c13a829af0dR83

but I just piggyback off of the TemplateService.

So, we still need to figure out the best way to do this